homeassistant/components/switch/onewire.py

```diff
@@ -0,0 +1,141 @@
+        if family in DEVICE_SWITCHES:
+            switch_id = os.path.split(
+                os.path.split(family_file_path)[0])[1]
+            for switch_key, switch_value in DEVICE_SWITCHES[family].items():
+                switch_id_complete = switch_id + "_" + switch_key
+                device_file = os.path.join(
+                    os.path.split(family_file_path)[0], switch_value)
+                devs.append(OneWireSwitch(device_names.get(switch_id_complete,
+                                                           switch_id_complete),
+                                          device_file))
+
+    if devs == []:
+        _LOGGER.error("No onewire sensor found. Check if dtoverlay=w1-gpio "
+                      "is in your /boot/config.txt. "
+                      "Check the mount_dir parameter if it's defined")
+        return
+
+    add_devices(devs, True)
+
+
+class OneWireSwitch(SwitchDevice):
+    """Representation of a OneWire switch."""
+
+    def __init__(self, name, device_file):
+        """Initialize the OneWire switch."""
+        self._name = name or DEVICE_DEFAULT_NAME
+        self._device_file = device_file
+        self._state = None
+
+    def _read_value_raw(self):
+        """Read the value as it is returned by the switch."""
+        try:
+            with open(self._device_file, 'r') as ds_device_file:
+                content = ds_device_file.read()
+        except FileNotFoundError:
+            msg = "1Wire switch file not found: %s", self._device_file
+            _LOGGER.warning(msg)
+        except OSError:
+            _LOGGER.warning("Error reading switch file %s", self._device_file)
+        return content
+
+    def _write_value_raw(self, value):
+        """Write the value to the switch."""
+        if value not in [DEVICE_SWITCH_ON, DEVICE_SWITCH_OFF]:
+            _LOGGER.error("Error in setting wrong value %s", value)
+        else:
+            try:
+                with open(self._device_file, 'w') as ds_device_file:
+                    ds_device_file.write(value)
+                return True
+            except FileNotFoundError:
+                msg = "1Wire switch file not found: %s", self._device_file
+                _LOGGER.warning(msg)
+            except OSError:
+                msg = "Error writing switch file %s", self._device_file
+                _LOGGER.warning(msg)
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    def update(self):
+        """Get the latest data from the switch and update the state."""
+        self._state = self._read_value_raw() == DEVICE_SWITCH_ON
+        return self._state
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        if self._write_value_raw(DEVICE_SWITCH_ON):
+            self.schedule_update_ha_state()
```

Comment:
Remove this too. It will be called automatically after the service call, since this is a polling entity.